### PR TITLE
use nightly toolchain for linting and formatting

### PR DESCRIPTION
**Pull request by danieleades**
makes sense to use the nightly toolchain for linting, though there's one caveat-

Clippy should really have a config file that specifies a MSRV (this project doesn't seem to specify one) in order to prevent it trying to apply lints that don't yet apply to stable (i've not seen this actually happen before, but in principle it's possible)

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -60,11 +60,11 @@ jobs:
         - name: Checkout sources
           uses: actions/checkout@v2.3.3
 
-        - name: Install stable toolchain
+        - name: Install nightly toolchain
```
Comment (ar3s3ru) [inline .github/workflows/rust.yml:63]: What's the advantage between the stable and the nightly toolchain for clippy?

**Review by ar3s3ru — COMMENTED** [review]
Hey, thanks for the PR!

Left a small comment 😄

File: .github/workflows/rust.yml (hunk 1 above)
Comment (danieleades) [reply .github/workflows/rust.yml:63]: Only that the nightly toolchain has a great deal more lints

Comment (danieleades) [reply .github/workflows/rust.yml:63]: Actually the more dramatic difference is in rustfmt- nightly fmt is far more powerful

**Pull request merged**